include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -118,9 +194,15 @@ template <
     requires matrix_concept<remove_cvref_t<trace_matrix_t>> &&
              std::Same<typename remove_cvref_t<trace_matrix_t>::entry_type, trace_directions>
 //!\endcond
-inline std::pair<std::list<gapped_database_t>, std::list<gapped_query_t>>
-alignment_trace(database_t && database, query_t && query, trace_matrix_t && matrix, alignment_coordinate const end_coordinate)
+inline std::pair<std::vector<gapped_database_t>, std::vector<gapped_query_t>>
+alignment_trace(
+    database_t && database,
+    query_t && query,
+    trace_matrix_t && matrix,
+    alignment_coordinate const end_coordinate,
+    std::optional<size_t> gapped_sequence_size = std::nullopt)
```

Comment:
This seems to be over engineered. Why not just take a vector fill it reverse and finally reverse move it element wise into the result. That is exactly the same complexity and does not require the statistics stuff and pre knowledge of the size etc.

Reply:
It needs to be touched any way, when we have gap structures.

Reply:
Yeah lets do this later!

Reply:
The thing is that you enforce to compute the `begin_coordinate` in align_result (i.e., traversing the trace matrix twice). Why not use the already know length of the `gapped_sequence`.

But, this has to be changed, because this only works for random access containers.